plugins/module_utils/acme/acme.py

```diff
@@ -38,6 +39,14 @@
     nopad_b64,
 )
 
+try:
+    import ipaddress
+except ImportError:
+    HAS_IPADDRESS = False
+    IPADDRESS_IMPORT_ERROR = traceback.format_exc()
+else:
+    HAS_IPADDRESS = True
```

Comment:
`traceback.format_exc()` is an overkill. Just do
```suggestion
try:
    import ipaddress
except ImportError as import_exc:
    IPADDRESS_IMPORT_ERROR = import_exc
else:
    IPADDRESS_IMPORT_ERROR = None
    del ipaddress  # prevent accidental use later in the code
```
Also, `HAS_IPADDRESS` is not really necessary as you can reuse the exception var to signal that it's missing.

Reply:
Since the code Felix used follows the https://docs.ansible.com/ansible/latest/dev_guide/developing_modules_best_practices.html#importing-and-using-shared-code, I would say that if we need a new way of doing things, we should first update the official docs.

Reply:
Makes sense. It's just that this is not applicable to every use-case.